File: 30DaysOfCode/day08_dictionariesAndMaps/dictionariesAndMaps.php
<?php
$_fp = fopen("php://stdin", "r");

fscanf($_fp, "%d", $N);
$tel = array();

for ($i = 0; $i < $N; $i++) {
    $str = trim(fgets($_fp));
    $strArr = explode(" ", $str);
    
    $tel[$strArr[0]] = $strArr[1];
}

while ($query = trim(fgets($_fp))) {
    if (isset($tel[$query])) {
        echo $query . "=" . $tel[$query] . "\n";
    } else {
        echo "Not found\n";
    }
}

?>

File: 30DaysOfCode/day08_dictionariesAndMaps/dictionariesAndMaps.rb
N = gets.chomp.to_i
tel = {}

(1..N).each do
    str = gets.strip
    strArr = str.split(' ')
    
    tel[strArr[0]] = strArr[1]
end

while query = gets
    query.strip!
    if tel.key? query
        print(query, "=", tel[query], "\n")
    else
        puts "Not found"
    end
end
